swarm/network/kademlia.go

```diff
@@ -369,6 +377,13 @@ func (k *Kademlia) Off(p OverlayConn) {
 		if k.addrCountC != nil {
 			k.addrCountC <- k.addrs.Size()
 		}
+		if k.nDepthC != nil {
```

Comment:
Just so I understand, why do we not need any `NotifyDepth` after this check?

Reply:
I think that they are serving a different purpose. NotifyDepth is notifying peers, but this channel is sending depth to some other part of the code for the same node. And we use it in the stream package for updating sync subscriptions. This is one of the purposes that TestSwarmNetwork is testing when adding and removing nodes.